src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -2061,6 +2162,19 @@ SubqueryPushdownMultiPlanTree(Query *queryTree)
 	 * in the logical optimizer.
 	 */
 	extendedOpNode = MultiExtendedOpNode(queryTree);
+
+	/*
+	 * Postgres standard planner converts having qual node to a list of and
+	 * clauses and expects havingQual to be of type List when executing the
```

Comment:
I think this comment needs a bit more clarification.

We should be mentioning that in some cases we use original query and in some casses we use the query that has passed through the standard planner. That's why we have this code here.